beacon_node/network/src/sync/range_sync/range.rs

```diff
@@ -141,13 +142,20 @@ where
                 debug!(self.log, "Finalization sync peer joined"; "peer_id" => %peer_id);
                 self.awaiting_head_peers.remove(&peer_id);
 
+                // Because of our change in finalized sync batch size from 2 to 1 and our transition
+                // to using exact epoch boundaries for batches (rather than one slot past the epoch
+                // boundary), we need to sync finalized sync to 2 epochs + 1 slot past our peer's
+                // finalized slot in order to finalize the chain locally.
```

Comment:
Yeah that's right! We needed to end the finalized chain sync to start head syncing! That was the reason. :)